### PR TITLE
Derive SLIP0044 coin type 42 by default.

### LINKED ISSUE #24: Still uses legacy cointype
We should update the tool to use the SLIP0044 coin type 42.

### PR DESCRIPTION
**Pull request by markusrichter**
Introduces option -legacy to generate and verify legacy coin type.

Fixes #24

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"github.com/decred/dcrd/dcrec/secp256k1"
 	"github.com/decred/dcrd/dcrutil"
 	"github.com/decred/dcrd/hdkeychain"
+	"github.com/decred/dcrwallet/wallet/udb"
```
Comment (jrick) [inline main.go]: Please don't import this, it will eventually become an internal package

Comment (markusrichter) [reply main.go]: What would be a better approach? Just copying the `CoinTypes` function over from the `udb` package? It is simple enough and I would rather not pull in all the dependencies from `udb`.

Comment (jrick) [reply main.go]: That's fine.  We should also get these numbers in the chainparams.  The wallet code was written with the idea of eventually adding them to the params, and there's a test that the output remains the same even if the parameters change.

Comment (markusrichter) [reply main.go]: I updated the PR with hardcoded coin type numbers. They should never be changed in the future again anyway, so this makes sure the code doesn't break if the chainparams are actually updated.

**Comment by markusrichter** [comment]
The PR has been adjusted for https://github.com/decred/dcrd/pull/1293.
After the dcrd PR has been merged I will fix the dependencies here.

**Comment by dajohi** [comment]
#33 made SLIP0044 the default

**Pull request closed without merging**